pkg/controller/servicemesh/webhooks/hacks.go

```diff
@@ -172,49 +170,41 @@ func createWebhookResources(ctx context.Context, mgr manager.Manager, log logr.L
 	return nil
 }
 
-func registerConversionWebhook(
+func RegisterConversionWebhook(
 	ctx context.Context,
-	mgr manager.Manager,
+	cl client.Client,
 	log logr.Logger,
 	operatorNamespace string,
 	path *string,
 	crdName string) error {
 
 	log.Info(fmt.Sprintf("Adding conversion webhook to %s CRD", crdName))
-	apixclient, err := clientapixv1.NewForConfig(mgr.GetConfig())
-	if err != nil {
+
+	crd := &apixv1.CustomResourceDefinition{}
+	if err := cl.Get(ctx, client.ObjectKey{Name: crdName}, crd); err != nil {
+		if errors.IsNotFound(err) {
```

Comment:
Do you think this would be better handled when we try registering the sme crd?  I mention this because registration only occurs for smcp at startup and if it's not there, we should probably fail.